packages/manager/src/views/WebPropertyEnvPage/components/AddMembers/AddMembers.tsx

```diff
@@ -0,0 +1,620 @@
+/* eslint-disable no-underscore-dangle */
+import { fetchRoverGroup, fetchUserlist, useAddPermission } from '@app/services/rbac';
+import { toPascalCase } from '@app/utils/toPascalConvert';
+import {
+  Accordion,
+  AccordionContent,
+  AccordionItem,
+  AccordionToggle,
+  Button,
+  Checkbox,
+  Flex,
+  FlexItem,
+  Label,
+  Select,
+  SelectOption,
+  SelectProps,
+  SelectVariant,
+  Split,
+  SplitItem,
+  Tab,
+  Tabs,
+  TabTitleText
+} from '@patternfly/react-core';
+import { InfoCircleIcon, UserIcon, UsersIcon } from '@patternfly/react-icons';
+import { TableComposable, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import { useRouter } from 'next/router';
+import React, { useEffect, useState } from 'react';
+import toast from 'react-hot-toast';
+import {
+  AddDataType,
+  ColumnNames,
+  NewRoverData,
+  RoverItem,
+  RoverUserList,
+  UserDataItem,
+  UserDataTDO,
+  UserRoleDTO
+} from './types';
+
+type Props = {
+  onClose: () => void;
+};
+
+const columnNames2: ColumnNames = {
+  NAME: 'Name',
+  APIKEY_CREATION: 'APIKEY_CREATION',
+  APIKEY_DELETION: 'APIKEY_DELETION',
+  PERMISSION_CREATION: 'PERMISSION_CREATION',
+  PERMISSION_DELETION: 'PERMISSION_DELETION',
+  ENV_CREATION: 'ENV_CREATION',
+  ENV_SYNC: 'ENV_SYNC'
+};
+const userRole: UserRoleDTO = {
+  User: ['APIKEY_CREATION', 'ENV_CREATION', 'ENV_SYNC'],
+  Owner: [
+    'APIKEY_CREATION',
+    'APIKEY_DELETION',
+    'ENV_CREATION',
+    'ENV_SYNC',
+    'PERMISSION_CREATION',
+    'PERMISSION_DELETION'
+  ]
+};
+
+export const AddMembers = ({ onClose }: Props): JSX.Element => {
+  const [activeTabKey, setActiveTabKey] = useState<string | number>(0);
+  const [isOpenUser, setIsOpen] = useState(false);
+  const [selectedUsers, setSelectedUsers] = useState<string[]>([]);
+  const [isShowAdvancedViewEnabled, setIsShowAdvancedViewEnabled] = useState(false);
+  const [isShowAdvancedViewEnabledRover, setIsShowAdvancedViewEnabledRover] = useState(false);
+  const titleId = 'plain-typeahead-select-id';
+  const [userListFromRover, setUserListFromRover] = useState<RoverUserList>([]);
+  const [usersData, setUsersData] = useState<UserDataTDO>({ data: [] });
+  const [newUserDetails, setNewUserDetails] = useState<NewRoverData[]>([]);
+  const [expanded, setExpanded] = useState('def-list-toggle2');
+  const { query } = useRouter();
+  const [roverList, setRoverList] = useState<NewRoverData[]>([]);
+  const propertyIdentifier = query.propertyIdentifier as string;
+
+  const handleTabClick = (
+    _event: React.MouseEvent<any> | React.KeyboardEvent | MouseEvent,
+    tabIndex: string | number
+  ) => {
+    setActiveTabKey(tabIndex);
+  };
+
+  const userDetailsString = JSON.stringify(newUserDetails);
+
+  useEffect(() => {
+    const userDetails: NewRoverData[] = activeTabKey === 0 ? newUserDetails : roverList;
```

Comment:
How about removing the newUser state entirely 

And instead when u want to add newUser just do this operation directly towards userData you can then remove this monitoring etc and keep it dead simple

Reply:
I did this part, but then I had to handle that state on every click action, which was being taken care by useEffect in terms of dependency array